docs/design.md

```diff
@@ -689,6 +698,40 @@ process is presented below.
 + Boot into image in the primary slot of the 0th image position\
   (other image in the boot chain is started by another image).
 
+### [Multiple Image Boot for RAM loading and direct-xip](#multiple-image-boot-for-ram-loading-and-direct-xip)
+
+The operation of the boot loader is different when the ram-load or the
+direct-xp strategy is chosen. The flash map is very similar to the swap
+strategy but there is no need for Scratch area.
+
++  Loop 1. Iterate over all images
+    + Does any of the slots contain an image?
+        + Yes:
+            + Choose the newer image.
+            + Copy it to RAM in case of ram-load stategy.
+            + Validate the image (integrity and security check).
+            + If validation fails delete the image from flash and try the other
+              slot. (Image must be deleted from RAM too in case of ram-load
+              stategy.)
+        + No: Return with failure.
+
++  Loop 2. Iterate over all images
```

Comment:
From the code, it seems that Loop 2 includes Loop 1 as a nested loop.  But the description here does not indicate this. Do I miss something?

Reply:
It was misleading, I updated the doc.